internal/controller/managedcluster_controller.go

```diff
@@ -303,24 +308,129 @@ func (r *ManagedClusterReconciler) Update(ctx context.Context, l logr.Logger, ma
 		requeue, err := r.setStatusFromClusterStatus(ctx, l, managedCluster)
 		if err != nil {
 			if requeue {
-				return ctrl.Result{RequeueAfter: 10 * time.Second}, err
+				return ctrl.Result{RequeueAfter: DefaultRequeueInterval}, err
 			}
 
 			return ctrl.Result{}, err
 		}
 
 		if requeue {
-			return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+			return ctrl.Result{RequeueAfter: DefaultRequeueInterval}, nil
 		}
 
 		if !fluxconditions.IsReady(hr) {
-			return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+			return ctrl.Result{RequeueAfter: DefaultRequeueInterval}, nil
 		}
+
+		return r.updateServices(ctx, managedCluster)
 	}
 
 	return ctrl.Result{}, nil
 }
 
+// updateServices reconciles services provided in ManagedCluster.Spec.Services.
+// TODO(https://github.com/Mirantis/hmc/issues/361): Set status to ManagedCluster object at appropriate places.
+func (r *ManagedClusterReconciler) updateServices(ctx context.Context, mc *hmc.ManagedCluster) (ctrl.Result, error) {
+	l := log.FromContext(ctx).WithValues("ManagedClusterController", fmt.Sprintf("%s/%s", mc.Namespace, mc.Name))
+	opts := []sveltos.HelmChartOpts{}
+
+	// NOTE: The Profile object will be updated with no helm
+	// charts if len(mc.Spec.Services) == 0. This will result in the
+	// helm charts being uninstalled on matching clusters if
+	// Profile originally had len(m.Spec.Sevices) > 0.
+	for _, svc := range mc.Spec.Services {
+		if svc.Disable {
+			l.Info(fmt.Sprintf("Skip adding Template (%s) to Profile (%s) because Disable=true", svc.Template, mc.Name))
+			continue
+		}
+
+		tmpl := &hmc.ServiceTemplate{}
+		tmplRef := types.NamespacedName{Name: svc.Template, Namespace: mc.Namespace}
+		if err := r.Get(ctx, tmplRef, tmpl); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to get Template (%s)", tmplRef.String())
+		}
+
+		source, err := r.getServiceTemplateSource(ctx, tmpl)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("could not get repository url: %w", err)
+		}
+
+		opts = append(opts, sveltos.HelmChartOpts{
+			RepositoryURL:  source.Spec.URL,
+			RepositoryName: tmpl.Spec.Helm.ChartName,
```

Comment:
Is it correct that the `RepositoryName` is set to the `ChartName`?

Reply:
The Sveltos docs use this format for OCI registries (scroll to the bottom): https://projectsveltos.github.io/sveltos/addons/helm_charts/

We are using OCI for now so it is ok. But good point, as I can check to see which type the HelmRepository is. If it is not OCI, then looking at Sveltos docs, the chartName needs to be in the format of `repository/chart` which in our case will become `ChartName/ChartName`, because we don't have a "RepositoryName" available in our spec.

Reply:
Pushed new commit to set these fields based on whether the source repository is of default or oci type.